test/CompressedBlockPrefilteringTest.cpp

```diff
@@ -0,0 +1,547 @@
+            blockIdx};
+  }
+
+  // Check if expected error is thrown.
+  auto makeTestErrorCheck(std::unique_ptr<PrefilterExpression> expr,
+                          const std::vector<BlockMetadata>& input,
+                          const std::string& expected,
+                          size_t evaluationColumn = 0) {
+    AD_EXPECT_THROW_WITH_MESSAGE(expr->evaluate(input, evaluationColumn),
+                                 ::testing::HasSubstr(expected));
+  }
+
+  // Check that the provided expression prefilters the correct blocks.
+  auto makeTest(std::unique_ptr<PrefilterExpression> expr,
+                std::vector<BlockMetadata>&& expected) {
+    std::vector<BlockMetadata> expectedAdjusted;
+    // This is for convenience, we automatically insert all mixed blocks
+    // which must be always returned.
+    std::ranges::set_union(
+        expected, mixedBlocks, std::back_inserter(expectedAdjusted),
+        [](const BlockMetadata& b1, const BlockMetadata& b2) {
+          return b1.blockIndex_ < b2.blockIndex_;
+        });
+    ASSERT_EQ(expr->evaluate(blocks, 0), expectedAdjusted);
+  }
+};
+
+}  // namespace
+
+//______________________________________________________________________________
+TEST_F(TestPrefilterExprOnBlockMetadata, testBlockFormatForDebugging) {
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: I:0 V:10 D:33.000000 V:0\n(last) "
+      "Triple: I:0 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b5).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: I:-4 V:10 D:33.000000 V:0\n(last) "
+      "Triple: D:2.000000 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b11).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: V:14 V:10 D:33.000000 V:0\n(last) "
+      "Triple: V:17 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b21).str());
+}
+
+// Test Relational Expressions
+//______________________________________________________________________________
+// Test LessThanExpression
+TEST_F(TestPrefilterExprOnBlockMetadata, testLessThanExpressions) {
+  makeTest(lt(IntId(5)),
+           {b5, b6, b9, b10, b11, b12, b13, b14, b15, b16, b17, b18});
+  makeTest(lt(IntId(-12)), {b18});
+  makeTest(lt(IntId(0)), {b9, b10, b15, b16, b17, b18});
+  makeTest(lt(DoubleId(-14.01)), {b18});
+  makeTest(lt(DoubleId(-11.22)), {b17, b18});
+  makeTest(lt(DoubleId(-4.121)), {b9, b15, b16, b17, b18});
+  makeTest(lt(VocabId(0)), {b18});
+  makeTest(lt(VocabId(12)), {b18, b19});
+  makeTest(lt(VocabId(14)), {b18, b19});
+  makeTest(lt(VocabId(16)), {b18, b19, b20, b21});
+  makeTest(lt(IntId(100)),
+           {b5, b6, b7, b8, b9, b10, b12, b13, b14, b15, b16, b17, b18});
+  makeTest(lt(undef), {});
+  makeTest(lt(falseId), {});
+  makeTest(lt(trueId), {b2, b3});
+  makeTest(lt(referenceDate1), {});
+  makeTest(lt(referenceDateEqual), {b22});
+  makeTest(lt(referenceDate2), {b22, b23, b24});
+  makeTest(lt(BlankNodeId(11)), {b24});
```

Comment:
I have a question:
The blocks `b4, b11, b18` all have mixed datatypes, I thus expect them to be part of ALL results.